app_dart/lib/src/service/luci.dart

```diff
@@ -102,7 +121,23 @@ class LuciService {
   }) async {
     assert(requireTaskName != null);
     final List<LuciBuilder> builders = await LuciBuilder.getProdBuilders(repo, config);
-    final Iterable<Build> builds = await getBuilds(repo, requireTaskName, builders);
+    final List<Build> builds = <Build>[];
+
+    // Request builders data in batches of 50 to prevent failures in the grpc service.
+    const RetryOptions r = RetryOptions(maxAttempts: 3);
+    for (int j = 0; j < builders.length; j += 50) {
+      final List<LuciBuilder> partialBuilders = builders.sublist(j, min(j + 49, builders.length - 1));
+      await r.retry(
+        () async {
+          final Iterable<Build> partialBuilds = await getBuilds(repo, requireTaskName, partialBuilders);
+          builds.addAll(partialBuilds);
+        },
+        retryIf: (Exception e) => e is BuildBucketException,
+      );
+      // Wait in between requests to prevent rate limiting.
+      final Random random = Random();
+      await Future<dynamic>.delayed(Duration(seconds: random.nextInt(10)));
```

Comment:
Nit: move this logic to a shared function?

Reply:
@keyonghan Actually I see the above function called from a single place would it be better to just remove it and updated https://cs.opensource.google/flutter/cocoon/+/master:app_dart/lib/src/request_handlers/push_engine_status_to_github.dart;l=51?q=getRecentTasks&ss=flutter%2Fcocoon to use the branched version.

Reply:
Update `push_engine_status_to_github` may not be an easy change.
`getRecentTasks` returns builds of a builder in descending order, which makes it easy to get the tot status for engine.
whereas `getBranchRecentTasks` returns a BranchLuciBuilder based map, which is then based on commit, and then LUCI tasks. The tot status of each builder needs a second calculation.

Reply:
Refactored to a method.